### PR TITLE
[script.program.fritzact] v0.0.17

### PR DESCRIPTION
**Pull request by b-jesch**
### Description
Control your DECT remote switches of the popular german Fritz!Box ADSL device within Kodi. Switch on or off your Illuminations or devices with just one click.
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.program.fritzact/addon.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿<?xml version="1.0" encoding="UTF-8"?>
+<addon id="script.program.fritzact" name="Fritz Smart Home" version="0.0.17" provider-name="Birger Jesch">
+    <requires>
+        <import addon="xbmc.python" version="2.24.0" />
+        <import addon="script.module.requests" version="2.9.1" />
+    </requires>
+	<extension point="xbmc.python.script" library="default.py">
```
Comment (Razzeee) [inline script.program.fritzact/addon.xml]: Just a tip, the smaller you can get `default.py` to be, the faster your addon will start up

Comment (b-jesch) [reply script.program.fritzact/addon.xml]: Is this a requirement or just a tip? I do not want to complain, but I've been waiting for a month now. And I have other projects within the pipeline too.

Comment (Razzeee) [reply script.program.fritzact/addon.xml]: Just a tip,  not blocking. I think nobody had any time to review this yet. :(

**Comment by b-jesch** [comment]
Sorry, is there anybody out there (why does come Pink Floyd through my mind)? Waiting now for two months for a reply...
Is there anything with the addon? Should I reject this?

Regards

File: script.program.fritzact/addon.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    </extension>
+
+    <extension point="xbmc.addon.metadata">
+        <platform>all</platform>
+        <summary lang="en">Switching your home with Kodi</summary>
+        <summary lang="de">Schalte Dein Zuhause mit Kodi</summary>
+        <description lang="en">Use your Fritz!DECT switches within Kodi. Control light and switch your devices with FritzDECT actors. Find out about powered-on devices and their power consumption. Make yourself comfortable with a touch of your remote.</description>
+        <description lang="de">Schalte Deine Fritz!DECT Steckdosen mit Kodi. Steuere Licht und Deine Geräte über FritzDECT Schalter. Informiere Dich über eingeschaltete Geräte und deren Stromverbrauch. Mach' es Dir bequem - mit nur einem Klick auf der Fernbedienung.</description>
+        <license>GNU GENERAL PUBLIC LICENSE. Version 3, June 2007</license>
+        <source>https://github.com/b-jesch/plugin.program.fritzact</source>
```
Comment (basrieter) [inline script.program.fritzact/addon.xml]: Seems that the sources url is giving a 404? Does it need updating?

Comment (b-jesch) [reply script.program.fritzact/addon.xml]: Indeed overseen that. At the moment I don't have access to my dev equipment. I'll correct this next week.

File: script.program.fritzact/resources/lib/tools.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+import xbmc
+import xbmcgui
+import xbmcaddon
+import os
+import random
+
+__addon__ = xbmcaddon.Addon()
+__addonID__ = __addon__.getAddonInfo('id')
+__path__ = __addon__.getAddonInfo('path')
```
Comment (basrieter) [inline script.program.fritzact/resources/lib/tools.py]: Same thing here: you could use `xbmc.translatePath(xbmcaddon.Addon().getAddonInfo('path')).decode('utf-8')` and then only use it with `os.path.join` for the other paths.

File: script.program.fritzact/default.py [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+import xbmcaddon
+import xbmcplugin
+from xml.etree import ElementTree as ET
+import re
+
+__addon__ = xbmcaddon.Addon()
+__addonID__ = __addon__.getAddonInfo('id')
+__addonname__ = __addon__.getAddonInfo('name')
+__version__ = __addon__.getAddonInfo('version')
+__path__ = __addon__.getAddonInfo('path')
```
Comment (basrieter) [inline script.program.fritzact/default.py]: you could use `xbmc.translatePath(xbmcaddon.Addon().getAddonInfo('path')).decode('utf-8')` here. That way you don't need to use `xbmc.translatePath` on all other locations that depend on __path__

**Comment by b-jesch** [comment]
Done. Suggestions of @basrieter implemented too.

**Comment by basrieter** [comment]
@b-jesch Great. Let me check why Travis CI is complain, but it does not seem to be related to this add-on.

**Pull request merged**